File: CLASS3/1697.py
from collections import deque
def bfs():
    q = deque()
    q.append((0, n))
    visit = [0 for _ in range(100001)]
    visit[n] = 1
    while q:
        cnt, now = q.popleft()
        if now == k:
            return cnt
        for i in (now-1, now+1, now*2):
            if 0 <= i <= 100000 and visit[i] == 0:
                visit[i] = 1
                q.append((cnt+1, i))

n, k = map(int, input().split())
result = bfs()
print(result)

File: solved.ac/CLASS3_JS/1697.js
const fs = require("fs");
const inputPath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const stdin = fs
  .readFileSync(inputPath)
  .toString()
  .split("\n")
  .map((line) => line.trim());

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const solution = () => {
  const [n, k] = input().split(" ").map(Number);

  const bfs = (start) => {
    const q = [[start, 0]];
    const visited = Array(100001).fill(100001);
    visited[start] = 0;

    while (q.length) {
      const [now, cnt] = q.shift();
      if (now === k) {
        return cnt;
      }
      for (const next of [now - 1, now + 1, now * 2]) {
        if (0 <= next && next < 100001 && visited[next] > cnt) {
          visited[next] = cnt;
          q.push([next, cnt + 1]);
        }
      }
    }
  };
  console.log(bfs(n));
};

solution();
